extensions/kafka-connect/src/main/java/com/hazelcast/jet/kafka/connect/KafkaConnectSources.java

```diff
@@ -71,13 +79,33 @@ public static StreamSource<SourceRecord> connect(@Nonnull Properties properties)
         //fail fast, required by lazy-initialized KafkaConnectSource
         checkRequiredProperty(properties, "connector.class");
 
-        return SourceBuilder.timestampedStream(name, ctx ->
-                        new KafkaConnectSource(properties))
-                .fillBufferFn(KafkaConnectSource::fillBuffer)
-                .createSnapshotFn(KafkaConnectSource::createSnapshot)
-                .restoreSnapshotFn(KafkaConnectSource::restoreSnapshot)
-                .destroyFn(KafkaConnectSource::destroy)
-                .build();
+        int tasksMax = Integer.parseInt(properties.getProperty("tasks.max", "1"));
+
+        return Sources.streamFromProcessorWithWatermarks(name, true,
+                eventTimePolicy -> ProcessorMetaSupplier.preferLocalParallelismOnSingleMember(
+                        new ProcessorSupplier() {
+                            private ConnectorWrapper connectorWrapper;
+
+                            @Override
+                            public void init(@Nonnull Context context) {
+                                this.connectorWrapper = new ConnectorWrapper(properties);
+                            }
+
+                            @Override
+                            public void close(@Nullable Throwable error) {
+                                connectorWrapper.stop();
```

Comment:
connectorWrapper may be null, see javadoc for close

Reply:
fixed: https://github.com/hazelcast/hazelcast/pull/23831/commits/18b1798181ec279cda16e2801225428b3f0806fa